### PR TITLE
Config file for SID mapping.

### PR DESCRIPTION
**Pull request by VorpalBlade**
Draft proof of concept support for custom SID mapping. Not yet even close to be ready for merging, but would love some feedback.

Open questions:
- [X] How to handle upgrades? Maybe a different file format instead of JSON would help with dpkg/UCF auto merging changes? How about other distros (e.g. Arch, Gentoo, etc)? - *With a YAML file this seems to work fairly well, invoking the usual config file collision when the package has a changed version of the file.*
- [X] Should the config file be world readable or is the SID mapping sensitive data? - *Not needed, UIDs and SIDs are public*
- [X] Would it be better to map from user name instead of UID?
- [X] Would it make sense to move the dev path parameter to the config file? - *Answer: No*

Work to be done:
- [X] Install default config file.
- [X] Documentation on this feature in the README.
- [X] Validation of config file. Might be nice using something like jsonschema (if we stick with JSON).

### CODE REVIEW

**Comment by VorpalBlade** [comment]
Example config file (as of now):
```json
{
  "uid_to_sid": {
    "1000": "S-1-5-21-1235678901-2345667890-1234567890-1234"
  }
}
```

**Comment by uunicorn** [comment]
> Would it make sense to move the dev path parameter to the config file?

If this is about `--devpath usb-<busnum>-<address>` parameter, I can't see how it could be added to a config file. 
While `busnum` is relatively stable, the `address` is just a sequential number which is incremented each time you plug something into USB port. The idea behind this parameter was to start as many DBus services (using udev rules) as we have supported devices plugged in. For that to work, I tried to use template systemd service file, but ran into life-cycle issues. Right now this parameter is unused.

**Comment by VorpalBlade** [comment]
> If this is about `--devpath usb-<busnum>-<address>` parameter, I can't see how it could be added to a config file.
> ...
Yes, and that answers my question. Good to know.

**Comment by VorpalBlade** [comment]
I changed to a YAML configuration file, as this allows for comments in the file, which is likely helpful to the user.

**Comment by uunicorn** [comment]
> Would it be better to map from user name instead of UID?

It seems to be more user-friendly. However, the default SID should probably be still generated with RID=UID, if no explicit mapping is available. Perhaps with something more sophisticated than `S-1-5-21-111111111-1111111111-1111111111-%d`. We should probably make a globally unique subauthority and reuse it for all SIDs generated on that host.

**Comment by VorpalBlade** [comment]
Everything is implemented, except we are still using S-1-5-21-111111111-1111111111-1111111111-%d as the default. Maybe basing it on /etc/machine-id could work?

File: debian/rules [hunk 1]
```diff
@@ -12,4 +12,4 @@ override_dh_installsystemd:
 
 override_dh_auto_install:
 	python3 ./setup.py install --root=$(CURDIR)/debian/tmp --prefix=/usr --install-layout=deb
-
+	install -Dm 644 etc/dbus-service.yaml $(CURDIR)/debian/tmp/etc/python-validity/dbus-service.yaml
```
Comment (uunicorn) [inline debian/rules:15]: IIRC, `.` (as well as `debian/tmp`) is a part of the search path  for installation candidates. So if you move/rename `etc/dbus-service.yaml` to `etc/python-validity/dbus-service.yaml` inside the project, this command will no longer be needed. Not 100% sure though.

**Comment by uunicorn** [comment]
> I think that could be handled as a separate PR though.

I guess we should start creating separate issues for all the TODOs and known bugs.

File: debian/rules (hunk 1 above)
Comment (VorpalBlade) [reply debian/rules]: Tested, that works

**Comment by VorpalBlade** [comment]
I have fixed all these issues, as well as added the required type annotations to make my IDE shout loudly at me if I were to mix up SidIdentity & User again (dynamic typing by default is probably the biggest weakness of Python in my opinion for "serious" projects)

**Comment by uunicorn** [comment]
Thanks!

**Pull request merged**